bc_obps/registration/models.py

```diff
@@ -494,10 +494,72 @@ class Statuses(models.TextChoices):
         blank=True,
         related_name="operations_and_facilities",
     )
+    tax_exemption_id = models.CharField(
+        max_length=255,
+        unique=True,
+        blank=True,
+        null=True,
+        db_comment="The tax exemption ID of an operation when operation is approved",
+    )
     history = HistoricalRecords(
         table_name='erc_history"."operation_history', m2m_fields=[regulated_products, reporting_activities, documents]
     )
 
+    def validate_tax_exemption_id(self):
+        existing_operations = self.__class__.objects.filter(
+            Q(tax_exemption_id=self.tax_exemption_id)
```

Comment:
Good use of complex query objects here it can be a good example as we can probably use them elsewhere in the code as well!